cs/src/Connections/TunnelClient.cs

```diff
@@ -240,6 +243,12 @@ protected async Task StartSshSessionAsync(Stream stream, CancellationToken cance
     private void OnSshSessionDisconnected(object? sender, EventArgs e) =>
         StartReconnectTaskIfNotDisposed();
 
+    private void OnSshSessionClosed(object? sender, EventArgs e)
+    {
+        ConnectionStatus = ConnectionStatus.Disconnected;
```

Comment:
Should we unsubscribe the event handler? Otherwise it'll pin the old `SshSession` object to `TunnelClient`

Reply:
And other handlers too.

Reply:
Done.